homeassistant/components/modbus/sensor.py

```diff
@@ -131,18 +131,22 @@ class SlaveSensor(
     RestoreEntity,
     SensorEntity,
 ):
-    """Modbus slave binary sensor."""
+    """Modbus slave register sensor."""
 
     def __init__(
         self,
         coordinator: DataUpdateCoordinator[list[int] | None],
         idx: int,
         entry: dict[str, Any],
     ) -> None:
-        """Initialize the Modbus binary sensor."""
+        """Initialize the Modbus register sensor."""
         idx += 1
         self._idx = idx
         self._attr_name = f"{entry[CONF_NAME]} {idx}"
+        self._attr_unique_id = entry.get(CONF_UNIQUE_ID)
```

Comment:
This is not allowed for core integrations. UNIQUE_ID must be set without configuration, because if configured the user can change it.

Reply:
Its already there on the parent, its just that the slaves where inconsistent and did not follow the parent unique id

Reply:
see:
https://www.home-assistant.io/integrations/modbus/
unique_id string (optional)
An ID that uniquely identifies this sensor. If two sensors have the same unique ID, Home Assistant will raise an exception.

Reply:
Ups forgot that long discussion we had before that commit was merged. You are right and adding it to slaves makes sense.